lib/MwbExporter/Formatter/Doctrine2/Annotation/Model/Column.php

```diff
@@ -112,24 +110,27 @@ public function writeGetterAndSetter(WriterInterface $writer)
                 $typehint = $shouldTypehintProperties && class_exists($nativeType) ? "?$nativeType " : '';
             }
 
-            $writer
-                // setter
-                ->write('/**')
-                ->write(' * Set the value of '.$this->getColumnName().'.')
-                ->write(' *')
-                ->write(' * @param '.$nativeType.' $'.$this->getColumnName())
-                ->write(' * @return '.$table->getNamespace())
-                ->write(' */')
-                ->write('public function set'.$this->getBeautifiedColumnName().'('.$typehint.'$'.$this->getColumnName().')')
-                ->write('{')
-                ->indent()
+            if(!$this->isPrimary) {
```

Comment:
I see. We had the same in natixis and we called that `externalFundId` as we didn't want BBH to influence the mechanism of the id assignment in the business object but at the same time we do want to keep track for their reference, and automatically mapping that to our identifiers. 

In general By declaring a field `providerId` as an extra property of the class we would have that covered for any case. At ORM level that would only differ from `->find(id)` to `->findBy([providerId => something]` ... and even in the case the main find should be on providerId a specific repository overwriting the standard find can be implemented so find would really do a findBy, even though through api filters and their relation with doctrine filters already provided, that would be easily achievable as GET `/someresource?providerId=something`

In the case where we couldn't really avoid and we would really really need to use external ids as our internal primary key, after a minute of silence, we could still achieve that by just re-adding the setter on the entity of interest. As far as I understand this is the code that would produce the initial scaffolding of the BO, but then we can add on and modify for specific non-standard cases.

I'm also ok to re-introduce the write for setId on the primary too, but imo if there is any setId() performed externally (exception given for the internal test framework) that could introduce side-effects as a client would be able to pass an id in the request payload at that point.

Reply:
another option is introducing an additional config were we can list entities that should have imported id and allow the setter to be written for those 
eg.
export.json
```
...
        "externalIdEntities": [
            "App\\Entity\\Planet",
            "App\\Entity\\SolarSystem"
        ],
...
```
but nevertheless I confirm my preferred way would always be use `providerId` as an extra independent column

Reply:
After discussion with @domgraziano we have agreed to a compromise where we will allow external id on the primary key only if an annotation allows it:
<d:external>true</d:external>
This would be a column description annotation. So the setter would be written if the column is a primary key AND <d:external>true</d:external>

Reply:
@peterchallett I added the discussed change requests here at d3ba4c9 
please note I've created also a `CustomComment` class that in my mind can holds reference as php constant to the custom comments we expect to receive from mysqlworkbench. 
We could include other old comment to that class as additional task at some point if you agree with the idea.

Reply:
@peterchallett fixed checked to be performed with `parseComment()` as reliable checker to be used at 0ad05cc

Reply:
@peterchallett updated 2f1cfbe with 
- avoid to return early 
- segregation of write get and set in their own private methods